### PR TITLE
Custom report export

### PR DESCRIPTION
**Pull request by andreabettich**
Give the possibility to configure the analysis export endpoints.
This PR is based on the #35 PR/Code, so please review #35 first.

### CODE REVIEW

File: app/analysis/controller.js [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
-  return (
-    base +
-    Object.keys(filters)
-      .map(key => {
-        return `${key}=${filters[key]}`
-      })
-      .join('&')
-  )
+const downloadUrl = (url, filters) => {
+  return `${url}&${Object.keys(filters)
```
Comment (czosel) [inline app/analysis/controller.js]: Looks mad, but is fine :smile:

File: app/analysis/controller.js [hunk 2]
```diff
@@ -22,10 +18,10 @@ export default Controller.extend({
   page: 1,
 
   sort: '-date',
-
+  exportLinks: config.APP.REPORTEXPORTS,
   actions: {
-    download(type, filters) {
-      let target = `${downloadUrl(filters)}&file_type=${type}`
+    download(url, filters) {
+      let target = downloadUrl(url, filters)
```
Comment (czosel) [inline app/analysis/controller.js]: What do you think about renaming the method to something like `appendFilters`?

Comment (andreabettich) [inline app/analysis/controller.js]: Yes, would be more informative.

Comment (andreabettich) [inline app/analysis/controller.js]: Changed it on #a582355

File: config/environment.js [hunk 3]
```diff
@@ -82,5 +82,21 @@ module.exports = function(environment) {
...
+  /* global process*/
+  try {
+    let additionalReportExports = JSON.parse(process.env.timed_report_export)
+    if (additionalReportExports) {
+      ENV.APP.REPORTEXPORTS = ENV.APP.REPORTEXPORTS.concat(
+        additionalReportExports
+      )
+    }
+  } catch (e) {
+    /* eslint no-console: ["error", { allow: ["error"] }] */
```
Comment (czosel) [inline config/environment.js]: There are easier ways to write this, for example:
`// eslint-disable-next-line no-console`

See http://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments

File: config/environment.js [hunk 4]
```diff
@@ -82,5 +82,21 @@ module.exports = function(environment) {
 
   }
 
+  /* global process*/
+  try {
+    let additionalReportExports = JSON.parse(process.env.timed_report_export)
+    if (additionalReportExports) {
+      ENV.APP.REPORTEXPORTS = ENV.APP.REPORTEXPORTS.concat(
```
Comment (czosel) [inline config/environment.js]: `ENV.APP.REPORTEXPORTS = [ ...ENV.APP.REPORTEXPORTS, ...additionalReportExports ]`

File: config/environment.js [hunk 5]
```diff
@@ -82,5 +82,21 @@ module.exports = function(environment) {
 
   }
 
+  /* global process*/
+  try {
+    let additionalReportExports = JSON.parse(process.env.timed_report_export)
+    if (additionalReportExports) {
```
Comment (czosel) [inline config/environment.js]: You might also want to check if it's an array.

**Comment by czosel** [comment]
I just discussed this a little further with @sliverc, we thought that the following should still be added:

When you develop locally, it should be easy to set up a custom environment. For that, we'd like to support setting the environment in an `.env` file. That file will not be checked in version control, but a command to set it up (`echo .. > .env`) should be added to the README in our private repo. In the README of this repo there should be some documentation on what this feature is doing and how to use it.

File: config/environment.js (hunk 4 above)
Comment (andreabettich) [inline config/environment.js]: Damn, that looks 1000x sexier :smile:

File: config/environment.js (hunk 3 above)
Comment (andreabettich) [inline config/environment.js]: Changed it

File: config/environment.js [hunk 6]
```diff
@@ -70,5 +82,24 @@ module.exports = function(environment) {
 
   }
 
+  /* global process*/
+  try {
+    let envReportExports = process.env.TIMED_REPORT_EXPORT
+    if (envReportExports) {
+      let additionalReportExports = JSON.parse(envReportExports)
+      if (additionalReportExports && additionalReportExports instanceof Array) {
```
Comment (czosel) [inline config/environment.js]: I think `Array.isArray()` is better practice than `instanceof`

Comment (andreabettich) [inline config/environment.js]: Didn't even know that existed. Nice!

**Pull request merged**